### PR TITLE
Fix interpolation boundaries

### PR DESCRIPTION
**Pull request by dschwoerer**
Keep workaround for lack of interpolation into boundary if converting to
fieldaligned
[Apply the hack for non-fieldaligned to fieldaligned](https://github.com/boutproject/BOUT-dev/compare/next...fix_interp_to?expand=1#diff-08115b2d2191f4b50484126770cfeb98L70)

Not sure this is the right thing to do, but at least fa and non-fa are this way consistent.

### CODE REVIEW

File: src/mesh/interpolation.cxx [hunk 1]
```diff
@@ -146,7 +146,10 @@ const Field3D interp_to(const Field3D &var, CELL_LOC loc, REGION region) {
           // coordinates
 
           Field3D var_fa = fieldmesh->toFieldAligned(var);
-          Field3D result_fa = fieldmesh->toFieldAligned(result);
+          Field3D result_fa(fieldmesh);
+          if (loc != RGN_NOBNDRY) {
```
Comment (dschwoerer) [inline src/mesh/interpolation.cxx]: shouldn't it be `region` rather then `loc`?

Comment (johnomotani) [reply src/mesh/interpolation.cxx]: yep, sorry. I must need a coffee...

File: src/mesh/interpolation.cxx [hunk 2]
```diff
@@ -146,7 +146,10 @@ const Field3D interp_to(const Field3D &var, CELL_LOC loc, REGION region) {
           // coordinates
 
           Field3D var_fa = fieldmesh->toFieldAligned(var);
-          Field3D result_fa;
+          Field3D result_fa(fieldmesh);
+          if (loc != RGN_NOBNDRY) {
+            result = fieldmesh->toFieldAligned(result);
```
Comment (dschwoerer) [inline src/mesh/interpolation.cxx]: and also `result_fa` rather then `result`?

Comment (johnomotani) [reply src/mesh/interpolation.cxx]: :+1: thanks @dschwoerer

**Pull request merged**